File: spring-web-java/src/main/java/spring/web/java/repository/custom/ItemCustomRepository.java
package spring.web.java.repository.custom;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import spring.web.java.domain.Item;

public interface ItemCustomRepository {

    Page<Item> findAllWithJpql(Pageable pageable, String name);

    Page<Item> findAllWithQuerydsl(Pageable pageable, String name);
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/item/repository/ItemCustomRepository.kt
package spring.web.kotlin.domain.item.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import spring.web.kotlin.domain.item.Item

interface ItemCustomRepository {
    fun findAllWithJpql(pageable: Pageable, name: String?): Page<Item>
}